# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.cpp
# Commit: 108d5fb4418e
# Full Hash: 108d5fb4418e78e4dca0a746863e5784c6d80371
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-23 09:52:48
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r d8dea951a032 -r 108d5fb4418e docshell/base/CanonicalBrowsingContext.cpp
--- a/docshell/base/CanonicalBrowsingContext.cpp	Wed Apr 22 15:48:20 2020 +0000
+++ b/docshell/base/CanonicalBrowsingContext.cpp	Wed Apr 22 15:48:17 2020 +0000
@@ -33,14 +33,14 @@
 #define USER_ACTIVATION_LOG(msg, ...) \
   MOZ_LOG(gUserInteractionPRLog, LogLevel::Debug, (msg, ##__VA_ARGS__))
 
-CanonicalBrowsingContext::CanonicalBrowsingContext(BrowsingContext* aParent,
+CanonicalBrowsingContext::CanonicalBrowsingContext(WindowContext* aParentWindow,
                                                    BrowsingContextGroup* aGroup,
                                                    uint64_t aBrowsingContextId,
                                                    uint64_t aOwnerProcessId,
                                                    uint64_t aEmbedderProcessId,
                                                    BrowsingContext::Type aType,
                                                    FieldTuple&& aFields)
-    : BrowsingContext(aParent, aGroup, aBrowsingContextId, aType,
+    : BrowsingContext(aParentWindow, aGroup, aBrowsingContextId, aType,
                       std::move(aFields)),
       mProcessId(aOwnerProcessId),
       mEmbedderProcessId(aEmbedderProcessId) {